streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractDualSchemaRocksDBSegmentedBytesStore.java

```diff
@@ -241,11 +241,10 @@ public String name() {
         return name;
     }
 
-    @Deprecated
     @Override
-    public void init(final ProcessorContext context,
-                     final StateStore root) {
-        this.context = context;
+    public void init(final StateStoreContext stateStoreContext, final StateStore root) {
+        this.stateStoreContext = stateStoreContext;
+        this.context = StoreToProcessorContextAdapter.adapt(stateStoreContext);
```

Comment:
This feels rather hacky -- using `adapt` was a workaround as long as we have both `init()` methods. I think, we need to do more cleanup/refactoring and get rid or `this.context` completely and use `this.stateStoreContext` only.

However, this PR is already rather larger, so it might be best to do a follow up PR for it.